frontend/lib/resolution/split-init.cpp

```diff
@@ -227,196 +235,161 @@ void FindSplitInits::handleYield(const uast::Yield* ast, RV& rv) {
-  VarFrame* thenFrame = currentThenFrame();
-  VarFrame* elseFrame = currentElseFrame();
-
-  // save results for vars declared in then/else
-  // gather the set of variables to consider
-  std::set<ID> locInitedVars;
-  if (thenFrame != nullptr) {
-    for (const auto& id : thenFrame->initedVars) {
-      if (thenFrame->eligibleVars.count(id) > 0) {
-        // variable declared in this scope, so save the result
-        allSplitInitedVars.insert(id);
-      } else if (elseFrame == nullptr ||
-                 elseFrame->mentionedVars.count(id) == 0) {
-        // variable inited in 'then' but not mentioned in 'else'
-        locInitedVars.insert(id);
-      }
-    }
-  }
-  if (elseFrame != nullptr) {
-    for (const auto& id : elseFrame->initedVars) {
-      if (elseFrame->eligibleVars.count(id) > 0) {
-        // variable declared in this scope, so save the result
-        allSplitInitedVars.insert(id);
-      } else if (thenFrame->mentionedVars.count(id) == 0) {
-        // variable inited in 'else' and not mentioned in 'then'
-        locInitedVars.insert(id);
-      }
+static void propogateMentions(VarFrame * parentFrame, VarFrame* frame) {
+  for (const auto& id : frame->mentionedVars) {
+    if (frame->declaredVars.count(id) == 0) {
+      parentFrame->mentionedVars.insert(id);
     }
   }
+}
 
-  bool thenReturnsThrows = false;
-  if (thenFrame != nullptr) {
-    thenReturnsThrows = thenFrame->returnsOrThrows;
-  }
-  bool elseReturnsThrows = false;
-  if (elseFrame != nullptr) {
-    elseReturnsThrows = elseFrame->returnsOrThrows;
+void FindSplitInits::errorChecking(const AstNode * node, 
+                                   std::vector<std::vector<ID>> idLists,
+                                   std::vector<std::vector<QualifiedType>> typeLists) {
+  for(size_t i = 1; i < idLists.size(); i++) {
+    CHPL_ASSERT(idLists.at(i).size() == 
+                idLists.at(i-1).size());
+    CHPL_ASSERT(typeLists.at(i).size()  ==
+                typeLists.at(i-i).size());
   }
 
-  std::set<ID> locSplitInitedVars;
-
-  // now consider the initialized 'vars' && propagate them
-  // split init is OK if:
-  //   both then & else blocks initialize it before mentioning it
-  //   one initializes it and the other returns
-  for (const auto& id : locInitedVars) {
-    bool thenInits = false;
-    if (thenFrame) {
-      thenInits = thenFrame->initedVars.count(id) > 0;
-    }
-    bool elseInits = false;
-    if (elseFrame) {
-      elseInits = elseFrame->initedVars.count(id) > 0;
-    }
+  bool orderOk = true;
 
-    if (thenInits && elseInits) {
-      locSplitInitedVars.insert(id);
-    } else if ((thenInits         && elseReturnsThrows) ||
-               (thenReturnsThrows && elseInits)) {
-      // one branch returns or throws and the other inits
-      locSplitInitedVars.insert(id);
-    } else {
-      frame->mentionedVars.insert(id);
+  std::vector<ID> referenceIds = idLists.at(0);
+  for (std::vector<ID> comparedIds : idLists) {
```

Comment:
The way this is written, you're always comparing `referenceIds` with itself. I assume this is unnecessary! Consider writing the outer loop to also start at 1 and iterate until `< idLists.size()`.